sdk/src/Services/DynamoDBv2/Custom/DocumentModel/DocumentTransactWrite.cs

```diff
@@ -325,6 +362,52 @@ internal void AddDocumentToUpdateHelper(Document document, Key key, TransactWrit
             });
         }
 
+        internal void AddKeyToUpdateHelper(Key key, Expression updateExpression, TransactWriteItemOperationConfig operationConfig)
+        {
+            if (operationConfig?.ConditionalExpression != null)
+            {
+                // Filter attribute names and values from the update expression that are also found in the condition expression
+                // If the value is different for the same key between the two expressions, an exception is thrown
+                // This ensures no exception is thrown when the attribute names and values of the update expression are merged into the request object
+                if (!TryFilterDuplicates(updateExpression.ExpressionAttributeValues,
+                        operationConfig.ConditionalExpression.ExpressionAttributeValues, out var attributeValues,
+                        out var conflictingValueKeys))
+                {
+                    throw new ArgumentException(
+                        "Conflicting attribute values found between the update expression and the condition expression. " +
+                        "Use a different key instead of each affected attribute value key in one of the expressions. " +
+                        $"Affected attribute value keys: {string.Join(", ", conflictingValueKeys)}.",
```

Comment:
For the two new `string.Join` calls here, can you add `ToArray()

We still target .NET Framework 3.5, which doesn't have the `Join` overload for a list of strings.
```
Severity	Code	Description	Project	File	Line	Suppression State	Details
Error	CS1503	Argument 2: cannot convert from 'System.Collections.Generic.List<string>' to 'string[]'	AWSSDK.DynamoDBv2.Net35	C:\dev\aws-sdk-net-staging\sdk\src\Services\DynamoDBv2\Custom\DocumentModel\DocumentTransactWrite.cs	379	Active	
``

Reply:
Done